components/sidebar/sidebar_category/sidebar_category.tsx

```diff
@@ -29,13 +32,23 @@ type Props = {
     };
 };
 
-export default class SidebarCategory extends React.PureComponent<Props> {
+type State = {
+    isMenuOpen: boolean;
+}
+
+export default class SidebarCategory extends React.PureComponent<Props, State> {
     categoryTitleRef: React.RefObject<HTMLButtonElement>;
+    categoryMenuRef: React.RefObject<SidebarCategoryMenuType>;
```

Comment:
Regardless of this being debug code, shouldn't this use the type wrapped with `mapStateToProps`? Unless `connect(SidebarCategoryMenu)` forwards calls to the `SidebarCategoryMenu`, I wouldn't think you could access fields of the wrapped component anyway.

Reply:
The problem here is actually `injectIntl`, since it doesn't return a wrapped class, it returns an object. Nonetheless, this is just used to type the object and since the class not wrapped by redux contains all the fields I need anyways (since we define the props and state in there), it works.

Reply:
In that case, I'm surprised it actually works though when the type here is set to the inner type, but the ref is attached to the wrapper.